crates/tools/src/scenario.rs

```diff
@@ -0,0 +1,174 @@
+use std::{
+    convert::Infallible,
+    path::{Path, PathBuf},
+    sync::Arc,
+    time::Instant,
+};
+
+use anyhow::Context;
+use edr_eth::remote::jsonrpc;
+use edr_evm::blockchain::BlockchainError;
+use edr_provider::{Logger, ProviderError, ProviderRequest};
+use flate2::bufread::GzDecoder;
+use indicatif::ProgressBar;
+use serde::Deserialize;
+use tokio::{runtime, task};
+
+#[derive(Clone, Debug, Deserialize)]
+struct ScenarioConfig {
+    provider_config: edr_provider::ProviderConfig,
+    logger_enabled: bool,
+}
+
+pub async fn execute(scenario_path: &Path, max_count: Option<usize>) -> anyhow::Result<()> {
+    let (config, requests) = load_requests(scenario_path).await?;
+
+    if config.logger_enabled {
+        // anyhow::bail!("This scenario expects logging, but logging is not yet
+        // implemented")
+    }
```

Comment:
Is this old code?

Reply:
Good catch thanks, that shouldn't be commented out. The scenario runner doesn't support logging yet (or maybe ever, idk how to do it relatively consistently), so I reenabled that line in [484e428](https://github.com/NomicFoundation/hardhat/pull/4928/commits/484e4288249abe91dcba4b213adb8b02762fc123)

Reply:
Why would there need to be special support for logging? It still executes the same RPC methods; it just writes logs to the CLI.

The scenario recorder doesn't have any guarantee about logging, right?

Reply:
Not sure how comparable that'll be across runs. Probably a solvable issue just not in this PR :)